### PR TITLE
EP-4012: moved creo catalog from openeo-geopyspark-driver and added i…

### PR DESCRIPTION
**Pull request by laxiwuka**
…mplementation for /validation

### CODE REVIEW

File: openeo_driver/dry_run.py [hunk 1]
```diff
@@ -532,6 +535,27 @@ def atmospheric_correction(self, method: str = None, *args) -> 'DriverDataCube':
     def mask_scl_dilation(self) -> 'DriverDataCube':
         return self._process("custom_cloud_mask", arguments={"method": "mask_scl_dilation"})
 
+    def get_missing_products(self):
+        missing_products = []
+
+        if self.metadata.get("id") == "SENTINEL2_L2A":
+            opensearch_endpoint = self.metadata.get("_vito", "data_source", "opensearch_endpoint")
```
Comment (soxofaan) [inline openeo_driver/dry_run.py:542]: I'm afraid that `self.metadata` will not always be usable/available.
From the dry run you should only trust the data from `DryRunDataTracer`

File: openeo_driver/dry_run.py [hunk 2]
```diff
@@ -532,6 +535,27 @@ def atmospheric_correction(self, method: str = None, *args) -> 'DriverDataCube':
...
+            if opensearch_endpoint and "creo" in opensearch_endpoint:
+                mission = CatalogConstants.missionSentinel2
+                level = CatalogConstants.level2A
+                catalog = CatalogClient(mission, level)
+                source_constraints = self._data_tracer.get_source_constraints()
+                temporal_extent = source_constraints[0][1]['temporal_extent']
```
Comment (soxofaan) [inline openeo_driver/dry_run.py:548]: you should not only look at the first element of `source_constraints`, but all

File: openeo_driver/dry_run.py [hunk 3]
```diff
@@ -532,6 +535,27 @@ def atmospheric_correction(self, method: str = None, *args) -> 'DriverDataCube':
     def mask_scl_dilation(self) -> 'DriverDataCube':
         return self._process("custom_cloud_mask", arguments={"method": "mask_scl_dilation"})
 
+    def get_missing_products(self):
+        missing_products = []
+
+        if self.metadata.get("id") == "SENTINEL2_L2A":
```
Comment (soxofaan) [inline openeo_driver/dry_run.py:541]: can we avoid this hardcoded collection id `SENTINEL2_L2A`?

File: openeo_driver/dry_run.py [hunk 4]
```diff
@@ -532,6 +535,27 @@ def atmospheric_correction(self, method: str = None, *args) -> 'DriverDataCube':
     def mask_scl_dilation(self) -> 'DriverDataCube':
         return self._process("custom_cloud_mask", arguments={"method": "mask_scl_dilation"})
 
+    def get_missing_products(self):
```
Comment (soxofaan) [inline openeo_driver/dry_run.py:538]: I wonder if it is possible to avoid defining this method on `DryRunDataCube`, 
not sure what alternatives there are

**Review by soxofaan — COMMENTED** [review]
sorry for delay, I just found some time for a quick review

File: openeo_driver/dry_run.py (hunk 1 above)
Comment (laxiwuka) [reply openeo_driver/dry_run.py]: I now pass the `_vito data_source` info to `DataSource.load_collection` so I can get that info from `source_constraints`.

File: openeo_driver/dry_run.py (hunk 2 above)
Comment (laxiwuka) [reply openeo_driver/dry_run.py]: I now look at all `load_collection` elements in the `source_constraints`.

File: openeo_driver/dry_run.py (hunk 3 above)
Comment (laxiwuka) [reply openeo_driver/dry_run.py]: Not sure how I can avoid this check.

File: openeo_driver/dry_run.py (hunk 4 above)
Comment (laxiwuka) [reply openeo_driver/dry_run.py]: As I don't need the metadata of the `DryRunDataCube` anymore, I moved this method to `DryRunDataTracer`. Is this a better place for it?

**Comment by soxofaan** [comment]
closing this issue.
Content of PR has been split in a part in Open-EO/openeo-python-driver and a part in Open-EO/openeo-geopyspark-driver in other branches (now merged in master)

**Pull request closed without merging**